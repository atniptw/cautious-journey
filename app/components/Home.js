// @flow
import { remote, ipcRenderer } from 'electron';

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import addCommit from '../actions/commit'
import { Link } from 'react-router-dom';
import styles from './Home.css';

type Props = {};

export default class Home extends React.Component {
  constructor(props) {
    super(props);
    this.state = { value: '' };

    this.openDialog = this.openDialog.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleMainProcessResponse = this.handleMainProcessResponse.bind(this);
    
    ipcRenderer.on('mainprocess-response', this.handleMainProcessResponse);
  }
  
  handleMainProcessResponse(event, commit) {
    const { addCommit } = this.props;
    addCommit(commit);
  }

  handleSubmit(event) {
    event.preventDefault();
    const data = { repo: this.state.value };
    ipcRenderer.send('request-mainprocess-action', data );
  }

  openDialog() {
    const repo = remote.dialog.showOpenDialog({
      properties: ['openDirectory'],
    })[0];

    this.setState(() => {
      return { value: repo };
    });
  }


  render() {
    return (
      <div>
        <div className={styles.container} data-tid="container">
          <h2>Code Analyzer</h2>
          <form onSubmit={this.handleSubmit}>
            <label>
              Repo:
              <input type="text" value={this.state.value}/>
            </label>
            <button onClick={this.openDialog}>Open</button>
          </form>
          <Link to="/commit">to Counter</Link>
        </div>
      </div>
    );
  }
}

Home.propTypes = {
  addCommit: PropTypes.func.isRequired
};
