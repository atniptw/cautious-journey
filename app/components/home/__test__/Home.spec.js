import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Home from '../Home';

import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

function setup() {
    const addCommit = jest.fn();
    return shallow(<Home addCommit={addCommit} />)
}

describe('Home component', () => {
    it('should render home component', () => {
        const component = setup();
        expect(component).toMatchSnapshot();
    });
});