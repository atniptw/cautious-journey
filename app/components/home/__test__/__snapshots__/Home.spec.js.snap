// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should render home component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    addCommit={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
        data-tid="container"
      >
        <h2>
          Code Analyzer
        </h2>
        <form
          onSubmit={[Function]}
        >
          <label>
            Repo:
            <input
              type="text"
              value=""
            />
          </label>
          <button
            onClick={[Function]}
          >
            Open
          </button>
        </form>
        <Link
          replace={false}
          to="/commit"
        >
          to Counter
        </Link>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Code Analyzer
          </h2>,
          <form
            onSubmit={[Function]}
          >
            <label>
              Repo:
              <input
                type="text"
                value=""
              />
            </label>
            <button
              onClick={[Function]}
            >
              Open
            </button>
          </form>,
          <Link
            replace={false}
            to="/commit"
          >
            to Counter
          </Link>,
        ],
        "className": "container",
        "data-tid": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Code Analyzer",
          },
          "ref": null,
          "rendered": "Code Analyzer",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                Repo:
                <input
                  type="text"
                  value=""
                />
              </label>,
              <button
                onClick={[Function]}
              >
                Open
              </button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Repo:",
                  <input
                    type="text"
                    value=""
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Repo:",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "type": "text",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Open",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Open",
              "type": "button",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "to Counter",
            "replace": false,
            "to": "/commit",
          },
          "ref": null,
          "rendered": "to Counter",
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
          data-tid="container"
        >
          <h2>
            Code Analyzer
          </h2>
          <form
            onSubmit={[Function]}
          >
            <label>
              Repo:
              <input
                type="text"
                value=""
              />
            </label>
            <button
              onClick={[Function]}
            >
              Open
            </button>
          </form>
          <Link
            replace={false}
            to="/commit"
          >
            to Counter
          </Link>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Code Analyzer
            </h2>,
            <form
              onSubmit={[Function]}
            >
              <label>
                Repo:
                <input
                  type="text"
                  value=""
                />
              </label>
              <button
                onClick={[Function]}
              >
                Open
              </button>
            </form>,
            <Link
              replace={false}
              to="/commit"
            >
              to Counter
            </Link>,
          ],
          "className": "container",
          "data-tid": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Code Analyzer",
            },
            "ref": null,
            "rendered": "Code Analyzer",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label>
                  Repo:
                  <input
                    type="text"
                    value=""
                  />
                </label>,
                <button
                  onClick={[Function]}
                >
                  Open
                </button>,
              ],
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Repo:",
                    <input
                      type="text"
                      value=""
                    />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Repo:",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "type": "text",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                ],
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Open",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Open",
                "type": "button",
              },
            ],
            "type": "form",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "to Counter",
              "replace": false,
              "to": "/commit",
            },
            "ref": null,
            "rendered": "to Counter",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
